db/data.py

```diff
@@ -460,3 +462,66 @@ def get_summary_data(mbid, offset=0):
         pass
 
     return summary
+
+def find_md5_duplicates(md5):
+    """Check if data comes from an audio file with the same md5 (Uses data->md5_encoded)
+        
+    Args: 
+        md5: The md5_encoded from the low-level data to check
+    
+    Returns:
+        The retrieved mbid if the md5 already exists
+        None if the md5 is not found
+    """
+
+    query = text("""
+                 SELECT lowlevel.mbid
+                 FROM lowlevel, lowlevel_json
+                 WHERE lowlevel_json.data#> '{metadata,audio_properties,md5_encoded}' ? :md5
+                 AND lowlevel.id = lowlevel_json.id
+                 """)
+    with db.engine.begin() as connection:
+        result = connection.execute(query, {"md5": md5})
+        row = result.fetchone()
+        if row:
+            # returns the retrieved mbid
+            return row[0]
+        else:
+            return None
+
+def generate_datasetitem_uuidv3():
+    """Generate a random UUIDv4 for dataset items that are not present in the DB as recordings
+    
+    Returns: 
+        itemuuid: The generated random UUIDv3 assigned to a specific dataset item
```

Comment:
The comment was wrong it generates UUIDv3.
The reaso to choose UUIDv3 is because it generates the same uuid with the same pair of namespace/string, it seems more consistent than a random uuid. I implemented both so we can choose the one that best fits project needs.